src/pages/CitiesPage/CitiesPage.jsx

```diff
@@ -1,18 +1,22 @@
 import Page from '../../components/common/Page/Page';
 import HeroSection from './HeroSection';
 import CitiesSection from './CitiesSection';
-import NotFoundSection from './NotFoundSection';
+import { SearchProvider } from '../../context/SearchContext';
+import { DataProvider } from '../../context/DataContext';
 
 const CitiesPage = () => {
   return (
-    <Page
-      title="Cities"
-      description="Discover comprehensive travel guides for cities around the world on Voyage. Explore detailed itineraries, local tips, and hidden gems in every city guide. Start your adventure and share your experiences with fellow travelers."
-    >
-      <HeroSection />
-      <CitiesSection />
-      {false && <NotFoundSection />}
-    </Page>
+    <SearchProvider>
```

Comment:
advise for the future, you can create HOC to wrap components without modifying them

e.g. `withSearchProvider(ComponentToWrap)`